pkg/controller/hawtio/hawtio_controller.go

```diff
@@ -337,9 +348,40 @@ func (r *ReconcileHawtio) Reconcile(request reconcile.Request) (reconcile.Result
 				_, err = r.coreClient.Secrets(request.Namespace).Create(ctx, certSecret, metav1.CreateOptions{})
 				reqLogger.Info("Client certificate created successfully", "secret", secretName)
 				if err != nil {
-					reqLogger.Error(err, "Creating the client certificate secret failed", err)
+					reqLogger.Error(err, "Creating the client certificate secret failed")
 					return reconcile.Result{}, err
 				}
+				//roll-out hawtio
+				err = triggerTheRolloutDeployment(ctx, r.client, request.NamespacedName, caSecret.ResourceVersion)
```

Comment:
My understanding is that we already perform a rollout deployment on config change at https://github.com/hawtio/hawtio-operator/blob/d9c52667bf7bf1b76341d424d75c0976673b2c40/pkg/resources/deployment.go#L103.

Would it be possible to reuse that mechanism by adding an annotation for the Secret, rather than triggering an extra rollout?

Reply:
well it didn't worked for me. It looks like deployment is updated by creating a new one so this label is overwritten.  Or I might did it in the wrong way.

Reply:
Yes, I think the new annotation must be passed down to the method that creates / updates the Deployment.